crates/sui-core/src/authority/authority_store.rs

```diff
@@ -2052,13 +2081,16 @@ impl ObjectStore for AuthorityStore {
 }
 
 /// A wrapper to make Orphan Rule happy
-pub struct ResolverWrapper<T: BackingPackageStore> {
-    pub resolver: Arc<T>,
+pub struct ResolverWrapper {
+    pub resolver: Arc<dyn BackingPackageStore + Send + Sync>,
```

Comment:
This will change from compile-time type to dynamic dispatch, which is slower in performance, right?
Any reasons for this kind of changes?

Reply:
As explained in person, `T` would have to be a `dyn BackingPackageStore` anyway, but its much harder to make the compiler happy with that arrangement.